hazelcast-oss/Dockerfile

```diff
@@ -1,13 +1,10 @@
 FROM alpine:3
 
-# Versions of Hazelcast
-ARG HZ_VERSION=5.5.0-SNAPSHOT
```

Comment:
> Can't delete this as its setup by Release pipeline

The release pipeline doesn't / shouldn't set this.
The image will now always build the _latest released_ version (which, today, would be `5.4.0`)

The release pipeline would be setting the version to `5.5.0-SNAPSHOT`, which isn't right most of the time going forwards.

The actions that do the release control what version is _actually_ shipped and I think that's how it already works.

> For EE Dockerfile, its set to blank so should do the same here for OSS

In the EE dockerfile they the `HZ_VERSION` is actually used, here it's not.

Reply:
In that case Release Pipeline would need to be updated as currently it updates HZ_VERSION in OSS and EE Dockerfiles. Or perhaps find a different approach 
Also, @ldziedziul can confirm this but OSS file update is used to determine if OSS image should be built and deployed
I think this will only effect next version of MAJOR, MINOR and BETA as they will branch from master

Reply:
Already discussed & addressed [here](https://github.com/hazelcast/hazelcast-docker/pull/747#discussion_r1599868041).

Reply:
@JackPGreen 

Please go ahead and remove HZ_VERSION from OSS

Can you reiterate as to why you need to keep the EE HZ_VERSION just  for ref?

So @ldziedziul and me had discussion about this offline in part because of 'how do we determine whether we should build/push OSS images' if HZ_VERSION is removed. 

So we have come up with a solution for this by way of creating a new file `.github/release_type` and then say what to build - OSS, EE or ALL. This way the Release Pipeline will not touch OSS/EE Dockerfile at all. I will post the summary in #dev-infra shortly

I will resolve this and approve PR

Reply:
> Can you reiterate as to why you need to keep the EE HZ_VERSION just for ref?

In OS, the version was only used to get the URL. We can get the URL other ways, so no need for the version.

In EE, the version is needed to set metadata. Even if we supply a version URL, we've no idea what the version of that is. So have we to keep it.
But we only _care_ about the version when it's a release, and there the proper version is supplied - so it can be empty by default.

Why did I refactor out the version? If there was a hardcoded version (that wasn't `5.5.0-SNAPSHOT`), _something_ would need to ensure that version is kept in sync each time it's released (the pipeline will be going `SNAPSHOT` -> `SNAPSHOT`, but this would be the previous released version to the next, so not covered).

Reply:
Thanks.